### PR TITLE
Feat/37/soldier personal data

### PR DESCRIPTION
**Pull request by ponomarovv**

### CODE REVIEW

File: MilitaryManager.Attachments.API/Controllers/AttachmentsController.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ public string GenerateNewDocument()
                 _logger.LogError(ex, $"Template for {templateName} document is not available");
             }
 
-            var unit = _unitService.GetUnitAsync(5).Result;
+            var unit = _unitService.GetUnitsByIdAsync(5).Result;
```
Comment (Alex-knu) [inline MilitaryManager.Attachments.API/Controllers/AttachmentsController.cs]: First - use await
Second - remove constant id

Comment (Alex-knu) [inline MilitaryManager.Attachments.API/Controllers/AttachmentsController.cs]: Use await, remove constant id and .Result

File: MilitaryManager.Attachments.API/Startup.cs [hunk 2]
```diff
@@ -58,6 +58,7 @@ public void ConfigureServices(IServiceCollection services)
                 c.SwaggerDoc("Attachments", new OpenApiInfo { Title = "Attachment", Version = "v1" });
             });
 
+            services.AddCustomUnitsServices();
```
Comment (Alex-knu) [inline MilitaryManager.Attachments.API/Startup.cs]: why?

File: MilitaryManager.Core/Services/DivisionService.cs [hunk 3]
```diff
@@ -40,11 +40,10 @@ public async Task<IEnumerable<DivisionDTO>> GetAllDivisionsListAsync()
 
         public async Task<DivisionDTO> GetDivisionByKeyAsync(int id)
         {
-            Specification<Division> specification = new Divisions.DivisionsList();
-            var division = await _divisionRepository.GetListBySpecAsync(specification);
-            var getDivision = division.FirstOrDefault(x => x.Id == id);
+            Specification<Division> specification = new Divisions.DivisionById(id);
```
Comment (Alex-knu) [inline MilitaryManager.Core/Services/DivisionService.cs]: rewrite like await _divisionRepository.GetFirstBySpecAsync(new Divisions.DivisionById(id));

File: MilitaryManager/client-app/src/app/components/finish-registration/finish-registration.component.ts [hunk 4]
```diff
@@ -75,22 +75,22 @@ export class FinishRegistrationComponent implements OnInit {
...
-      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="Розмір голови")?.id, 0, this.commanderForm.get("headSize")?.value, null));
-      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="Розмір протигазу")?.id, 0, this.commanderForm.get("gasMaskSize")?.value, null));
-      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="Тип форми")?.id, 0, this.commanderForm.get("uniform")?.value, null));
-      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="Група крові")?.id, 0, this.commanderForm.get("bloodType")?.value, null));
-      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="Вага")?.id, 0, this.commanderForm.get("weight")?.value, null));
-      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="Зріст")?.id, 0, this.commanderForm.get("height")?.value, null));
+      newUnit.profiles.push(new ProfileModel(0, this.attributes.find(x => x.name=="����� ����")?.id, 0, this.commanderForm.get("footSize")?.value, null));
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/components/finish-registration/finish-registration.component.ts]: ?

File: MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/unit-info/unit-info.component.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+
+  ngOnInit(): void {
+    this.showFullUnitInfo();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    console.log('ngOnChanges')
+    console.log(changes);
+    // console.log(changes[this.idChild1]);
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/unit-info/unit-info.component.ts]: remove coments

File: MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/unit-info/unit-info.component.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+          this.parentFullName = `${this.unit?.parent?.lastName} ${this.unit?.parent?.firstName} ${this.unit?.parent?.secondName}`;
+        } else {
+          this.parentFullName = 'Немає';
+        }
+
+        this.divisionName = this.unit?.division.name;
+
+
+        console.log(this.unit)
+        // console.log(this.unit.profiles[0].name);
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/unit-info/unit-info.component.ts]: coments

File: MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/units-list/units-list.component.scss [hunk 7]
```diff
@@ -1,4 +1,4 @@
-@use "src/app/components/test1/test1.component";
+@use "../../../../../components/test1/test1.component";
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/units-list/units-list.component.scss]: ?

File: MilitaryManager/client-app/src/app/modules/units-module/pages/units-list/units-list.component.html [hunk 8]
```diff
@@ -1,9 +0,0 @@
-<div class="content-section implementation p-d-flex" id="container">
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/modules/units-module/pages/units-list/units-list.component.html:1]: Why you delete files?

File: MilitaryManager/client-app/src/app/modules/units-module/pages/units-list/units-list.component.ts [hunk 9]
```diff
@@ -1,52 +0,0 @@
-import { Component, OnInit } from '@angular/core';
-import { UnitModel } from 'src/app/shared/models/unit.model';
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/modules/units-module/pages/units-list/units-list.component.ts:2]: Why you delete files?

File: MilitaryManager/client-app/src/app/shared/models/unit.model.ts [hunk 10]
```diff
@@ -1,29 +1,50 @@
...
+import {RankModel} from "./rank.model";
+import {PositionModel} from "./position.model";
 
 export class UnitModel extends BaseModel {
-  public parent: UnitModel | null = null;
+  // public lastName: string | null = null;
+  // public secondName: string | null = null;
+  // public label: string | null = null;
+  // public firstName: string | null = null;
+  // public divisionId: number | null = null;
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/shared/models/unit.model.ts]: remove coments

File: MilitaryManager/client-app/src/app/shared/services/api/unit-user.service.ts [hunk 11]
```diff
@@ -17,6 +17,7 @@ export class UnitUserService extends BaseService<any>{
   }
 
   GetUnitUser(id:string) {
+    console.log(`${this.single.baseUrl}${id}`);
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/shared/services/api/unit-user.service.ts]: remove

File: MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts [hunk 12]
```diff
@@ -15,4 +15,8 @@ export class UnitsService extends BaseService<any> {
     configService: ClientConfigurationService) {
     super(httpService, 'unit', configService, UnitModel, ServiceType.units);
   }
+
+  GetUnitUser(id: number) {
+    return this.httpService.get(`${this.single.baseUrl}unit/units/${id}`);
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts]: we dont have such a controller in the developer

File: MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/unit-tree-and-info.component.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+
+  public id: number = 9;
+
+  public transferIdFromUnitListToMainComponent(someNumber: number): void {
+    console.log(this.id)
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/modules/units-module/pages/unit-tree-and-info/unit-tree-and-info.component.ts]: remove

**Review by Alex-knu — CHANGES_REQUESTED** [review]

File: MilitaryManager.Attachments.API/Controllers/AttachmentsController.cs (hunk 1 above)
Comment (ponomarovv) [reply MilitaryManager.Attachments.API/Controllers/AttachmentsController.cs]: it's not about this task at all).

**Review by Zhykovskiy — APPROVED** [review]

File: MilitaryManager.Attachments.API/Startup.cs (hunk 2 above)
Comment (ponomarovv) [reply MilitaryManager.Attachments.API/Startup.cs]: ok. i removed it. fixed

File: MilitaryManager.Attachments.API/MilitaryManager.Attachments.API.csproj [hunk 14]
```diff
@@ -1,45 +1,45 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
-  <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
-    <UserSecretsId>994a9585-5b8c-4de2-b12f-57d75623946d</UserSecretsId>
-    <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
-    <DockerComposeProjectPath>..\docker-compose.dcproj</DockerComposeProjectPath>
-  </PropertyGroup>
+	<PropertyGroup>
```
Comment (Alex-knu) [inline MilitaryManager.Attachments.API/MilitaryManager.Attachments.API.csproj:3]: undo

**Review by Alex-knu — CHANGES_REQUESTED** [review]

File: MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts [hunk 15]
```diff
@@ -1,9 +1,9 @@
-import { Injectable } from "@angular/core";
-import { UnitModel } from "../../models/unit.model";
-import { HttpService } from "../core/http.service";
-import { BaseService } from "../core/base.service";
-import { ClientConfigurationService } from "../core/client-configuration.service";
-import { ServiceType } from "../core/serviceType";
+import {Injectable} from "@angular/core";
+import {UnitModel} from "../../models/unit.model";
```
Comment (Alex-knu) [inline MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts:2]: undo

**Review by Alex-knu — CHANGES_REQUESTED** [review]

**Review by Alex-knu — APPROVED** [review]

**Pull request merged**